modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2363,6 +2364,47 @@ def drop(self, index=None, columns=None, errors: str = "raise"):
 
     # END Drop/Dropna
 
+    def duplicated(self, **kwargs):
+        def _compute_hash(df):
+            return df.apply(
+                lambda s: hashlib.new("md5", str(tuple(s)).encode()).hexdigest(), axis=1
+            ).to_frame()
+
+        def _compute_duplicated(df):
+            return df.duplicated(**kwargs).to_frame()
+
+        new_index = self._modin_frame._index_cache
+        new_columns = [MODIN_UNNAMED_SERIES_LABEL]
+        if len(self.columns) > 1:
```

Comment:
we can submit `.duplicated` right away if there's only one column partition, we don't need to squash them into a single one
```suggestion
        if self._modin_frame._partitions.shape[1] > 1:
```

Reply:
We "squash" the partitions along axis=0, but not axis=1, for the final phase. Also, if we checked for the number of partitions you propose, that would harm performace because there might be empty partitions.

```python
# perf with your suggestion
time for modin: 2.48 s
```

Reply:
could you share the script you're using to measure this?

Reply:
The script is the same as in the issue with NRUNS=1.

Reply:
You're right, this causes a slowdown as in this case the `column_widths_cache` is being None for the resulting frame thus causing it to be manually materialized further in the `drop_duplicates` flow.

However, as we're producing a Series here we know ahead of time that the column widths would be `[1]`, so if we'll be able to provide this hint the results became:
```
NPARTITIONS=112
PR 5587: 1.13s.
PR + my suggestion: 0.21s.

NPARTITIONS=8
PR 5587: 1.85s.
PR + my suggestion: 0.13s.
```

<details><summary>script</summary>

```python
import modin.pandas as pd
import pandas
import numpy as np
import timeit
from asv_bench.benchmarks.utils.common import execute

# def exetuce(df):
#     pass

NROWS = 1_000_000
RAND_LOW = 0
RAND_HIGH = 10_000_000

data = {
    "col0": np.tile(
        np.concatenate([np.random.randint(RAND_LOW, RAND_HIGH, NROWS // 100 - 1), [1]]),
        100,
    ),
    "col1": np.tile(
        np.concatenate([np.random.randint(RAND_LOW, RAND_HIGH, NROWS // 100 - 1), [1]]),
        100,
    ),
}

md_df = pd.DataFrame(data)
pd_df = pandas.DataFrame(data)

# drop_duplicate: 
#   input_shape = (1_000_000, 2)
#   output_shape = (10_000, 2)

NRUNS = 1
execute(md_df.sum()) # warmup
md_res = timeit.timeit(lambda: execute(md_df.drop_duplicates()), number=NRUNS)
pd_res = timeit.timeit(lambda: pd_df.drop_duplicates(), number=NRUNS)

print(f"Cumulative time of {NRUNS} runs for modin: {md_res}s.") # ~10s.
print(f"Cumulative time of {NRUNS} runs for pandas: {pd_res}s.") # ~0.3s
```

</details>

The improvement is quite significant to try investigating possibilities of how we could pass this hint to the modin frame. It could be either explicitly via parameter or we can try to precompute it inside `.apply_full_axis` based on the fact that `len(new_columns) == 1`. What do you think?

Reply:
I think we can start passing a hint on new lengths/widths to the core modin frame. I was already thinking about this a long time ago. Done this way.